### PR TITLE
docs (cli, core, schema): add architecture files

### PR DESCRIPTION
**Pull request by xavdid**
Inspired by https://matklad.github.io/2021/02/06/ARCHITECTURE.md.html, I wanted to write up some files to give new contributors a brief tour of the code.

[PDE-1990](https://zapierorg.atlassian.net/browse/PDE-1990)

### CODE REVIEW

File: packages/cli/ARCHITECTURE.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+- The `core`, `schema`, and `cli` packages are always released together under matching version numbers.
+
+## Technical Organization
+
+> `cli/...`, when used in a path to a file, is shorthand for `zapier-platform/packages/cli/...`
+
+### README
+
+- For legacy reasons, the tooling for the `zapier-platform/README.md` lives in this folder.
+- Changes to the `README` should made in `cli/README-source.md` and propagated using the `yarn docs` command in the `cli` folder. That command mostly performs text replacements in `README-source.md` for the latest released version number, supported Node.js version, and snippet text
```
Comment (eliangcs) [inline packages/cli/ARCHITECTURE.md:16]:
```suggestion
- Changes to the `README` should be made in `cli/README-source.md` and propagated using the `yarn docs` command in the `cli` folder. That command mostly performs text replacements in `README-source.md` for the latest released version number, supported Node.js version, and snippet text
```

File: packages/core/ARCHITECTURE.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# `zapier-platform-core` Architecture
+
+## Purpose
+
+- This package, released as `zapier-platform-core` publicly on [npm](https://www.npmjs.com/package/zapier-platform-core), ships code that partners (literally) depend on.
+- Once declared as a dependency and installed, developers can import it to help with testing & TypeScript types.
+- It's not actually `require`d directly in app code, but it runs an app's exposed functions at runtime.
+- Its main responsibilities are:
+  - organizing data passed from the Zapier monolith into the developer code (see []())
```
Comment (eliangcs) [inline packages/core/ARCHITECTURE.md:9]: Looks like a missing link?

File: packages/legacy-scripting-runner/ARCHITECTURE.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# `zapier-platform-legacy-scripting-runner` Architecture
+
+## Purpose
+
+- This package, released as `zapier-platform-cli` publicly on [npm](https://www.npmjs.com/package/zapier-platform-legacy-scripting-runner), is a compatibility shim between the legacy v2 platform and the current "v3" (aka CLI) platform.
```
Comment (eliangcs) [inline packages/legacy-scripting-runner/ARCHITECTURE.md:5]:
```suggestion
- This package, released as `zapier-platform-legacy-scripting-runner` publicly on [npm](https://www.npmjs.com/package/zapier-platform-legacy-scripting-runner), is a compatibility shim between the legacy v2 platform and the current "v3" (aka CLI) platform.
```

File: packages/legacy-scripting-runner/ARCHITECTURE.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+- The lion's share of code lives in `runner/index.js`.
+- The primary purposes are to:
+  - make available global functions that were available in the legacy environment (see the `compileLegacyScriptingSource` function)
+  - handle `pre_X` and `post_X` lifecycle hooks (see `runEventCombo` function)
+  - ensure inputs and outputs continue to allow what was allowed in the legacy environment (while CLI apps may have more strict requirements, such as needing to return arrays from triggers)
+- There are also individual files in `runner/*` that power functions/libraries exposed above, such as `btoa`, `$` (for `jQuery`), etc.
+
+### Tests
+
+- The main test file is `runner/test/integration-test.js`, which takes the basic example app from `runner/test/example-app`, tweaks a single trigger/action at a time, and makes asserts about how it performs.
```
Comment (eliangcs) [inline packages/legacy-scripting-runner/ARCHITECTURE.md:25]: I'd mention `runner/test/example-app/index.js` to be more specific:

```suggestion
- The main test file is `runner/test/integration-test.js`, which takes the basic example app from `runner/test/example-app/index.js`, tweaks a single trigger/action at a time, and makes asserts about how it performs.
```

**Review by eliangcs — DISMISSED** [review]
I love these docs, plain and simple! I only have some minor suggestions...

**Comment by xavdid** [comment]
cool, fixed in fec19f63! Glad you like them 😁

**Review by eliangcs — APPROVED** [review]
💯

**Pull request merged**